network/wsPeer.go

```diff
@@ -169,6 +169,12 @@ const disconnectStaleWrite disconnectReason = "DisconnectStaleWrite"
 const disconnectDuplicateConnection disconnectReason = "DuplicateConnection"
 const disconnectBadIdentityData disconnectReason = "BadIdentityData"
 
+const lastSentRequestTimeKey string = "lsrt"
+
+// If the peer sends us a block response after this threshold
+// we should disconnect from it
+const blockResponseDisconnectThreshold = 60 * time.Second
```

Comment:
That's a valid point, we can increase this or drop it. 

Alternatively we could potentially make it a protocol validation by checking our receive time and not sending anything out if we know that the requesting peer is guaranteed to not be expecting it any longer.

Reply:
the code below will fail to serve the request anyway since the handler to write response to is removed in 4 seconds
```
childCtx, cancelFunc := context.WithTimeout(ctx, time.Duration(w.config.CatchupGossipBlockFetchTimeoutSec)*time.Second)
```
maybe 4 sec is too short for full blocks

Reply:
Agreed and removed any timestamp logic in favor of counting number of topic requests we've sent to the peer. 

Should we change the timeout as part of this PR?